examples/python/upcloud-server/main.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+
+from glob import glob
+from importlib.metadata import metadata
+import os
+
+from constructs import Construct
+from cdktf import App, TerraformOutput, TerraformStack
+
+from imports.upcloud.provider import UpcloudProvider
+from imports.upcloud.server import Server, ServerLogin, ServerNetworkInterface, ServerTemplate
+
+
+def get_server_login(user=None):
+    """Get ServerLogin with current users first public key
+
+    As a fallback, create password if current user does not have any public
+    keys under ~/.ssh directory.
+    """
+    user_ssh_key_paths = glob(os.path.expanduser("~/.ssh/*.pub"))
+
+    if user_ssh_key_paths:
+        with open(user_ssh_key_paths[0]) as f:
+            ssh_key = f.read()
+        return ServerLogin(user=user, keys=[ssh_key])
+    else:
+        return ServerLogin(user=user, create_password=True, password_delivery="email")
```

Comment:
I like that this function tries to give a good default for users by reading their first (and oftentimes only) SSH key – But as we synthesize our examples in CI, I'm not too fond of having code that contains read access to the SSH directory of the Github Action runner that this is running in (even if it doesn't actually do something with it, other than writing it to the output `cdk.tf.json` template).

I'd prefer this to default to the password based `ServerLogin` and only try to read from `~/.ssh` if either a boolean flag, or even the name of the ssh public key file has been passed.

Reply:
Thanks for the quick review! I'll modify this part so that SSH key is only read if user gives the name of the ssh key as variable